auth0/src/main/java/com/auth0/android/dpop/DPoPUtil.kt

```diff
@@ -0,0 +1,259 @@
+            Log.e(TAG, "Error signing data: ${e.stackTraceToString()}")
+            throw DPoPException(DPoPException.Code.SIGNING_ERROR, e)
+        }
+    }
+
+    private fun convertDerToRawSignature(derSignature: ByteArray): ByteArray {
+        // DER format: SEQUENCE (0x30) + length + INTEGER (0x02) + length + R + INTEGER (0x02) + length + S
+        var offset = 0
+        if (derSignature[offset++] != 0x30.toByte()) throw SignatureException("Invalid DER signature: Expected SEQUENCE")
+        val length = decodeLength(derSignature, offset).also { offset += it.second }.first
+        if (length + offset != derSignature.size) throw SignatureException("Invalid DER signature: Length mismatch")
+
+        if (derSignature[offset++] != 0x02.toByte()) throw SignatureException("Invalid DER signature: Expected INTEGER for R")
+        val rLength = decodeLength(derSignature, offset).also { offset += it.second }.first
+        var r = derSignature.copyOfRange(offset, offset + rLength)
+        offset += rLength
+
+        if (derSignature[offset++] != 0x02.toByte()) throw SignatureException("Invalid DER signature: Expected INTEGER for S")
+        val sLength = decodeLength(derSignature, offset).also { offset += it.second }.first
+        var s = derSignature.copyOfRange(offset, offset + sLength)
+        offset += sLength
+
+        // Remove leading zero if present
+        if (r.size > 1 && r[0] == 0x00.toByte() && (r[1].toInt() and 0x80) == 0x80)
+            r = r.copyOfRange(1, r.size)
+        if (s.size > 1 && s[0] == 0x00.toByte() && (s[1].toInt() and 0x80) == 0x80)
+            s = s.copyOfRange(1, s.size)
+
+        // Pad with leading zeros to 32 bytes for P-256
+        val rawR = ByteArray(32)
+        System.arraycopy(r, 0, rawR, 32 - r.size, r.size)
+        val rawS = ByteArray(32)
+        System.arraycopy(s, 0, rawS, 32 - s.size, s.size)
+
+        return rawR + rawS
+    }
+
+    private fun decodeLength(data: ByteArray, offset: Int): Pair<Int, Int> {
+        var len = data[offset].toInt() and 0xFF
+        var bytesConsumed = 1
+        if ( (len and 0x80) != 0) {
+            val numBytes = len and 0x7F
+            len = 0
+            for (i in 0 until numBytes) {
+                len = (len shl 8) or (data[offset + 1 + i].toInt() and 0xFF)
+            }
+            bytesConsumed += numBytes
+        }
+        return Pair(len, bytesConsumed)
+    }
+
+    private fun cleanUrl(url: String): String {
+        return try {
+            val uri = java.net.URI(url)
+            val cleanedUri = java.net.URI(
+                uri.scheme,
+                uri.userInfo,
+                uri.host,
+                uri.port,
+                uri.path,
+                null, // Remove query
+                null  // Remove fragment
+            )
+            cleanedUri.toString()
+        } catch (e: Exception) {
+            Log.w(TAG, "Failed to parse URL, using original: $url", e)
+            url
```

Comment:
Should this method just throw? After all, the server will reject a DPoP proof with an `htu` that does not match the request's URL.